core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1524,23 +1524,24 @@ private static RelDataType arrayCompactReturnType(SqlOperatorBinding opBinding)
   private static RelDataType arrayInsertReturnType(SqlOperatorBinding opBinding) {
     final RelDataType arrayType = opBinding.collectOperandTypes().get(0);
     final RelDataType componentType = arrayType.getComponentType();
-    final RelDataType elementType = opBinding.collectOperandTypes().get(2);
+    final RelDataType elementType1 = opBinding.collectOperandTypes().get(1);
```

Comment:
This function collectOperandTypes() is called 3 times. If it allocates a new list, the list should probably be reused.